### PR TITLE
Structure to import CSV Items

### PR DESCRIPTION
**Pull request by gabrielfs7**
The necessary abstractions to have the CSV item import and be prepared for different template definitions.

https://oat-sa.atlassian.net/browse/ADF-105

### CODE REVIEW

File: model/import/Parser/CsvParser.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoQtiItem\model\import\Validator\HeaderValidator;
+use oat\taoQtiItem\model\import\Validator\LineValidator;
+use oat\taoQtiItem\model\import\Validator\ValidatorInterface;
+use oat\taoQtiItem\model\import\TemplateInterface;
+
+class CsvParser extends ConfigurableService implements ParserInterface
+{
+    public function parse(string $fileContent, TemplateInterface $template): array
```
Comment (boajer) [inline model/import/Parser/CsvParser.php:33]: Do we know expected size of `csv` files? Might be it worth to look into direction of streams parsing or generators?

Comment (gabrielfs7) [reply model/import/Parser/CsvParser.php:33]: We do not know yet, but we have a constraint to small files, cause for this phase the process not be handled in multiple workers. I will take a look when I reach that part, thanks! :)

**Review by boajer — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by habemuscode — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Comment by voknil** [comment]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by voknil — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**